app/Models/User.php

```diff
@@ -348,19 +349,45 @@ public function registerMediaCollections(): void
         $this->registerAvatarMediaCollection();
     }
 
+    // == actions
+
+    /**
+     * @return Builder<User>
+     */
+    public static function whereName(?string $displayNameOrUsername): Builder
+    {
+        if ($displayNameOrUsername === null) {
+            return static::query();
+        }
+
+        return static::query()
+            ->where(function ($query) use ($displayNameOrUsername) {
+                $query->where(function ($q) use ($displayNameOrUsername) {
+                    $q->whereNotNull('display_name')
+                      ->where('display_name', '!=', '')
+                      ->where('display_name', $displayNameOrUsername);
```

Comment:
doesn't "where displayname=value" eliminate the need for the other two checks?

Reply:
Strange, `->where('display_name', '!=', '')` appears to be eliminated in the branch head. I don't see that clause in latest.

Reply:
It's gone now, but my question remains. Can't this (365-369) be simplified to:
```
          $query->where('display_name', $displayNameOrUsername)
                ->orWhere('User', $displayNameOrUsername);
```

Reply:
Yep, it can. Updated in latest.